cmd/probe/internal/binding/streaming_replication.go

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright (C) 2022-2023 ApeCloud Co., Ltd
+
+This file is part of KubeBlocks project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package binding
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/kit/logger"
+	"k8s.io/utils/env"
+)
+
+const (
+	streamingReplicatorPort = 3502
+	xbstreamBinary          = "/usr/bin/xbstream"
+	xtrabackupBinary        = "/usr/bin/xtrabackup"
+)
+
+var (
+	logReplicator = logger.NewLogger("streaming.replication")
+
+	dbHost     = "127.0.0.1"
+	dbUser     = env.GetString("KB_SERVICE_USER", "root")
+	dbPassword = env.GetString("KB_SERVICE_PASSWORD", "")
+	dataDir    = env.GetString("KB_SERVICE_DATA_DIR", "/data/mysql/data")
+)
+
+func init() {
+	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", streamingReplicatorPort))
+	if err != nil {
+		panic(fmt.Sprintf("streaming replicator listen error: %s", err.Error()))
+	}
+	logReplicator.Info(fmt.Sprintf("streaming replicator is listening at %s...", listener.Addr().String()))
+	go streamingReplicator(listener)
+}
+
+func (ops *BaseOperations) StreamingReplicationOps(ctx context.Context, req *bindings.InvokeRequest, rsp *bindings.InvokeResponse) (OpsResult, error) {
```

Comment:
If this is a light-weight disk populating API, please design a complete populating API for component.

Reply:
It will be addressed in new component definition API.